talib/_abstract.pxi

```diff
@@ -399,6 +401,12 @@ class Function(object):
         # restore opt_input values to as they were before this call
         for param_name, value in opt_input_values:
             self.__opt_inputs[param_name]['value'] = value
+        self.__info['parameters'] = self.parameters
```

Comment:
`set_function_args` mutates `self.__info['parameters']` and `self.__info['input_names']` so those should be restored as well